### PR TITLE
MAYA-121074 fix assertion during edit-as-maya undo / redo

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- The addition of the item to the pull set was not undone / redone.
- The import already has its own undo /redo but we were also using a UsdUndoBlock, which meant USD modifiaction were undone / redone twice.

### CODE REVIEW

**Comment by pierrebai-adsk** [comment]
In the end I was able to fix all assertions.

File: lib/mayaUsd/fileio/primUpdaterManager.cpp [hunk 1]
```diff
@@ -172,6 +185,7 @@ bool writePullInformation(const Ufe::Path& ufePulledPath, const MDagPath& path)
     // Store medata on DG node
     auto              ufePathString = Ufe::PathString::string(ufePulledPath);
     MFnDependencyNode depNode(path.node());
+    MStatus           status;
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/fileio/primUpdaterManager.cpp:188]: FYI: the previous version of the code was re-suing a variable that is now inside the lambda.

File: lib/mayaUsd/fileio/primUpdaterManager.cpp [hunk 2]
```diff
@@ -288,9 +302,6 @@ PullImportPaths pullImport(
         return PullImportPaths(addedDagPaths, pulledUfePaths);
     }
 
-    // Record all USD modifications in an undo block and item.
-    UsdUndoBlock undoBlock(&UsdUndoableItemUndoItem::create("Pull import USD data modifications"));
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/fileio/primUpdaterManager.cpp:292]: This is the redundant UsdUndoBlock that was improperly capturing what the import was doing. The importer has its own undo/redo capture system.

**Review by ppt-adsk — APPROVED** [review]
Thanks!  Bit of a shame that manual inspection of test output was required to find these, would have been nice to get outright test failure.

**Pull request merged**